src/kirby/components/icon/custom-icon-name.directive.ts

```diff
@@ -0,0 +1,43 @@
+import { Directive, ElementRef, Input, Inject } from '@angular/core';
+
+import { CUSTOM_FONT_SETTINGS, CustomIconSettings } from './custom-icon-settings';
+
+@Directive({
+  // tslint:disable-next-line:directive-selector
+  selector: 'kirby-icon[customName]',
+})
+export class CustomIconNameDirective {
+  @Input() set customName(name: string) {
+    if (name) {
+      this.handleCustomIcon(name);
+    }
+  }
+
+  constructor(
+    private element: ElementRef,
+    @Inject(CUSTOM_FONT_SETTINGS) private customIconSettings: CustomIconSettings
+  ) {}
+
+  handleCustomIcon(name: string) {
+    if (this.customIconSettings[0].icons) {
+      const icon = this.getCustomIcon(name);
+
+      if (icon !== undefined) {
+        this.setCustomIcon(icon.svg);
+      } else {
+        console.warn('Icon with name', name, 'was not found in custom font.');
+      }
+    }
+  }
+
+  getCustomIcon(name: string): { name: string; svg: string; unicode: string } {
```

Comment:
Same comment as in the `tns` version of the directive, you may what to mark the private functions as private, just to make it easier to read the code. 😄